src/type-translator.ts

```diff
@@ -164,51 +185,78 @@ export class TypeTranslator {
   }
 
   translate(type: ts.Type): string {
-    // See the function `buildTypeDisplay` in the TypeScript compiler source
-    // for guidance on a similar operation.
-
-    // NOTE: type.flags is a single value for primitive types, but sometimes a
-    // bitwise 'or' of some values for more complex types.  We use a switch
-    // statement for the basics and a series of "if" tests for the complex ones,
-    // roughly in the same order the flags occur in the TypeFlags enum.
+    // NOTE: Though type.flags has the name "flags", it can only be one of
+    // the enum options at a time.  This switch handles all the cases in
+    // the ts.TypeFlags enum in the order they occur.
     switch (type.flags) {
       case ts.TypeFlags.Any:
         return '?';
       case ts.TypeFlags.String:
+      case ts.TypeFlags.StringLiteral:
         return 'string';
       case ts.TypeFlags.Number:
+      case ts.TypeFlags.NumberLiteral:
         return 'number';
       case ts.TypeFlags.Boolean:
+      case ts.TypeFlags.BooleanLiteral:
         return 'boolean';
+      case ts.TypeFlags.Enum:
+      case ts.TypeFlags.EnumLiteral:
+        return 'number';
+      case ts.TypeFlags.ESSymbol:
+        // NOTE: currently this is just a typedef for {?}, shrug.
+        return 'symbol';
       case ts.TypeFlags.Void:
         return 'void';
       case ts.TypeFlags.Undefined:
         return 'undefined';
       case ts.TypeFlags.Null:
         return 'null';
-      case ts.TypeFlags.EnumLiteral:
-      case ts.TypeFlags.Enum:
-        return 'number';
-      case ts.TypeFlags.StringLiteral:
-        return 'string';
-      case ts.TypeFlags.BooleanLiteral:
-        return 'boolean';
-      case ts.TypeFlags.NumberLiteral:
-        return 'number';
+      case ts.TypeFlags.Never:
+        this.warn(`should not emit a 'never' type`);
```

Comment:
The question is what we want to do when we hit one of these.  E.g. if you accidentally make an array of never, what type do we emit for Closure?

These warnings are never shown to a user and only show up inside the test suite, so it's probably harmless to leave it in... (?)